src/redux/slices/notifications.ts

```diff
@@ -0,0 +1,205 @@
+    }),
+    setAllAsSeen: (state: NotificationsState) => {
+      const newPagedNotifications = {};
+      const newNotifications = [];
+      Object.keys(state.paginatedNotifications).forEach((pageKey) => {
+        const old = {
+          ...state.paginatedNotifications[pageKey],
+          data: [...state.paginatedNotifications[pageKey].data],
+        };
+
+        const newData = old.data.map((notification) => ({
+          ...notification,
+          seen: true,
+        }));
+        newPagedNotifications[pageKey] = {
+          ...old,
+          data: newData,
+        };
+        newNotifications.push(...newData);
+      });
+
+      return {
+        ...state,
+        unseenCount: 0,
+        notifications: newNotifications,
+        paginatedNotifications: newPagedNotifications,
+      };
+    },
+    setAllAsRead: (state: NotificationsState) => {
+      const newPagedNotifications = {};
+      const newNotifications = [];
+      Object.keys(state.paginatedNotifications).forEach((pageKey) => {
+        const old = {
+          ...state.paginatedNotifications[pageKey],
+          data: [...state.paginatedNotifications[pageKey].data],
+        };
+
+        const newData = old.data.map((notification) => ({
+          ...notification,
+          read: true,
+        }));
+        newPagedNotifications[pageKey] = {
+          ...old,
+          data: newData,
+        };
+        newNotifications.push(...newData);
+      });
+
+      return {
+        ...state,
+        notifications: newNotifications,
+        paginatedNotifications: newPagedNotifications,
+      };
+    },
+    setNotificationAsRead: (
+      state: NotificationsState,
+      action: PayloadAction<{ messageId: string }>,
+    ) => {
+      const newPagedNotifications = {};
+      const newNotifications = [];
+
+      Object.keys(state.paginatedNotifications).forEach((pageKey) => {
+        const old = {
+          ...state.paginatedNotifications[pageKey],
+          data: [...state.paginatedNotifications[pageKey].data],
+        };
+        const newData = old.data.map((notification) => {
+          // eslint-disable-next-line no-underscore-dangle
+          if (notification._id === action.payload.messageId) {
+            return {
+              ...notification,
+              read: true,
+            };
+          }
+          return notification;
+        });
+        newPagedNotifications[pageKey] = {
+          ...old,
+          data: newData,
+        };
+        newNotifications.push(...newData);
+      });
+
+      return {
+        ...state,
+        notifications: newNotifications,
+        paginatedNotifications: newPagedNotifications,
+      };
+    },
+    setDeleteNotification: (
+      state: NotificationsState,
+      action: PayloadAction<{ messageId: string }>,
+    ) => {
+      const newPagedNotifications = {};
+      const newNotifications = [];
+      Object.keys(state.paginatedNotifications).forEach((pageKey) => {
+        const old = {
```

Comment:
I noticed the bits 
```
const old = {
          ...state.paginatedNotifications[pageKey],
          data: [...state.paginatedNotifications[pageKey].data],
        };
```
 and 

```
newPagedNotifications[pageKey] = {
          ...old,
          data: newData,
        };
        newNotifications.push(...newData);
```
Are repeated multiple times but the middle part changes, is there a better way to keep the code DRY?